File: packages/gbnf-experiment/data/20260909T040730Z_0947f229/ported_implementation/gbnf/rules_builder/char_at.py
"""Indexing helper.

JavaScript returns ``undefined`` for an out-of-range string index, and the
reference implementation leans on that heavily (``src[pos] === '"'``,
``if (src[pos])``). Python raises instead, so every such read goes through this
helper, which yields the empty string — falsy, and equal to no character.
"""

from __future__ import annotations


def char_at(src: str, pos: int) -> str:
    if 0 <= pos < len(src):
        return src[pos]
    return ''

File: proposed-projects/round-trip-experiment/reverse/20260909T040730Z_0947f229/20260910T065554Z_5d586b52/ported_implementation/src/rules-builder/char-at.ts
/**
 * Indexing helper.
 *
 * Out-of-range string indexing yields `undefined`, and the parser leans on that
 * heavily (`src[pos] === '"'`, `if (src[pos])`). This helper narrows that to the
 * empty string — falsy, and equal to no character — so the comparisons stay
 * total.
 */

export const charAt = (src: string, pos: number): string =>
  pos >= 0 && pos < src.length ? src[pos] : '';
